File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    var answer = 0;
    var array = [];
    while(section.length > 0){
        let del = section[0]+m-1;
        while(section.length > 0 && section[0] <= del){
        section.splice(0,1);
            
        }
        answer++;
    }
    return answer;
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
import java.util.*;

class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        int meter = 0;
        
        Arrays.sort(section);
        
        for(int i=0; i<section.length; i++){
            if(section[i]<=meter) continue;
            
            meter = section[i]+m-1;
            answer++;
        }
        return answer;
    }
}